sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -289,58 +287,66 @@ public async ValueTask<string> GetHeaderValueAsync(HttpMessage message, TokenReq
             private (TaskCompletionSource<HeaderValueInfo> InfoTcs, TaskCompletionSource<HeaderValueInfo>? BackgroundUpdateTcs, bool GetTokenFromCredential)
                 GetTaskCompletionSources(TokenRequestContext context)
             {
+                var localState = _state;
+                // Check if the current state requires no updates to _state under lock and is valid.
+                // All checks must be done on localState as it can't be modified by any other threads.
+                if (localState.InfoTcs != null && !RequestRequiresNewToken(context, localState.CurrentContext) && localState.InfoTcs.Task.IsCompleted && localState.BackgroundUpdateTcs == null)
```

Comment:
You can add `Deconstruct` to `TokenRequestState` to use individual variables for each state item

`RequestRequiresNewToken` can be checked last:
```cs
infoTcs != null && backgroundUpdateTcs == null && infoTcs.Task.Status == TaskStatus.RanToCompletion && !RequestRequiresNewToken(context, currentContext)
```

Reply:
I also used Deconstruct inside of the lock, let me know if you'd prefer to keep it as it was.

Reply:
After moving functionality to Properties/Methods I have reverted the Deconstruct, let me know if you think it still makes sense to have it.